Lib/asyncio/streams.py

```diff
@@ -802,3 +1028,672 @@ def __aiter__(self):
+                'message': ('An open stream object is being garbage '
+                            'collected; call "stream.close()" explicitly.')
+            }
+            if self._source_traceback:
+                context['source_traceback'] = self._source_traceback
+            self._loop.call_exception_handler(context)
+            transport.abort()
+        else:
+            self._reject_connection = True
+        self._stream_wr = None
+
+    @property
+    def _stream(self):
+        if self._stream_wr is None:
+            return None
+        return self._stream_wr()
+
+    def connection_made(self, transport):
+        if self._reject_connection:
+            context = {
+                'message': ('An open stream was garbage collected prior to '
+                            'establishing network connection; '
+                            'call "stream.close()" explicitly.')
+            }
+            if self._source_traceback:
+                context['source_traceback'] = self._source_traceback
+            self._loop.call_exception_handler(context)
+            transport.abort()
+            return
+        super().connection_made(transport)
+        stream = self._stream
+        if stream is None:
+            return
+        stream.set_transport(transport)
+        stream._protocol = self
+
+    def connection_lost(self, exc):
+        super().connection_lost(exc)
+        self._stream_wr = None
+
+
+class _ServerStreamProtocol(_BaseStreamProtocol):
+    def __init__(self, server, limit, client_connected_cb, loop=None,
+                 *, _asyncio_internal=False):
+        super().__init__(loop=loop, _asyncio_internal=_asyncio_internal)
+        assert self._closed
+        self._client_connected_cb = client_connected_cb
+        self._limit = limit
+        self._server = server
+        self._task = None
+
+    def connection_made(self, transport):
+        super().connection_made(transport)
+        stream = Stream(mode=StreamMode.READWRITE,
+                        transport=transport,
+                        protocol=self,
+                        limit=self._limit,
+                        loop=self._loop,
+                        is_server_side=True,
+                        _asyncio_internal=True)
+        self._stream = stream
+        # If self._client_connected_cb(self._stream) fails
+        # the exception is logged by transport
+        self._task = self._loop.create_task(
+            self._client_connected_cb(self._stream))
+        self._server._attach(stream, self._task)
+
+    def connection_lost(self, exc):
+        super().connection_lost(exc)
+        self._server._detach(self._stream, self._task)
+        self._stream = None
+
+
+def _swallow_unhandled_exception(task):
+    # Do a trick to suppress unhandled exception
+    # if stream.write() was used without await and
+    # stream.drain() was paused and resumed with an exception
+
+    # TODO: add if not task.cancelled() check!!!!
```

Comment:
Need to clean this up before committing.

Reply:
Done